traits/observers/_has_traits_helpers.py

```diff
@@ -27,6 +31,53 @@
 ]
 
 
+def warn_comparison_mode(object, name):
+    """ Check if the trait is a Dict/List/Set with comparison_mode set to
+    equality (or higher). If so, warn about the fact that observers will be
+    lost in the event of reassignment.
+
+    Parameters
+    ----------
+    object : HasTraits
+        Object where a trait is defined
+    name : str
+        Name of a trait to check if warning needs to be issued for it.
+    """
+    ctrait = object.traits()[name]
+
+    def has_container_trait(ctrait):
```

Comment:
Please pull this out to a module-level function; I don't think there's any need for it to be a nested function here.